src/test/java/reposense/model/RepoLocationTest.java

```diff
@@ -1,49 +1,116 @@
+        assertValidLocation(LOCAL_REPO_LOCATION_VALID_WITH_DOT_GIT_ONE,
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION);
+    }
+
+    @Test
+    public void repoLocation_parseWindowsLocalRepoLocation_success() throws Exception {
+        Assume.assumeTrue(SystemUtil.isWindows());
+        // repeated tests but with windows file separators
+        assertValidLocation(LOCAL_REPO_LOCATION_WINDOWS_VALID_WITHOUT_DOT_GIT_ONE,
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION);
+        assertValidLocation(LOCAL_REPO_LOCATION_WINDOWS_VALID_WITHOUT_DOT_GIT_TWO,
+                EXPECTED_REPO_NAME, "..-" + EXPECTED_ORGANIZATION);
+        assertValidLocation(LOCAL_REPO_LOCATION_WINDOWS_VALID_WITH_DOT_GIT_ONE,
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION);
+
+        assertValidLocation(LOCAL_REPO_LOCATION_WINDOWS_VALID_MIXED,
+                EXPECTED_REPO_NAME, "..-" + EXPECTED_ORGANIZATION);
+    }
+
+    @Test
+    public void repoLocation_parseValidRemoteRepoUrl_success() throws Exception {
+        // valid url without specifying branch
+        assertValidLocation("https://github.com/reposense/testrepo-Beta.git",
+                "testrepo-Beta", "reposense");
+        assertValidLocation("https://github.com/reposense/testrepo-Delta.git",
+                "testrepo-Delta", "reposense");
+        assertValidLocation("https://gitlab.com/reposense/RepoSense.git",
+                "RepoSense", "reposense");
+        assertValidLocation("https://github.com/reposense.git",
+                "reposense", "");
+        // valid url to parse for obtaining repo and organization, just not a valid git clone target
+        assertValidLocation("https://github.com/reposense/.git",
+                "reposense", "");
+
+        // treated as valid but will be caught when git clone fails
+        assertValidLocation("https://github.com/reposense/testrepo-Beta/tree/add-config-json",
+                "add-config-json", "reposense-testrepo-Beta-tree");
+        assertValidLocation("https://github.com/reposense/testrepo-Beta.git/tree/add-config-json",
+                "add-config-json", "reposense-testrepo-Beta.git-tree");
+
+        // URLs without ".git" should be accepted as git clone works even without it
+        assertValidLocation("https://github.com/reposense",
+                "reposense", "");
+        assertValidLocation("https://github.com/reposense/RepoSense",
+                "RepoSense", "reposense");
+
+        // Test against other types of URL protocols that are valid for git clone
+        assertValidLocation("ssh://[user@]host.xz[:port]/path/to/repo.git/",
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION);
+        assertValidLocation("git://host.xz[:port]/path/to/repo.git",
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION);
+        assertValidLocation("https://host.xz[:port]/path/to/repo.git",
+                EXPECTED_REPO_NAME, EXPECTED_ORGANIZATION);
```

Comment:
Are we able to change the `[user@]` and `[:port]` to a concrete example?

Reply:
The idea I had in mind was a kind of white-box testing where the domain is dropped regardless. But noted, I will try to change the examples here. Thanks!